File: answers/leetcode/Reverse Integer/Reverse Integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """

        reversed = 0
        sign = (x >= 0) * 2 - 1
        x = abs(x)
        while x:
            reversed = reversed * 10 + x % 10
            x /= 10
        reversed *= sign
        if reversed > 2147483647 or reversed < - 2147483648:
            reversed = 0
        return reversed

File: answers/leetcode/Reverse Integer/Reverse Integer.java
class Solution {
    public int reverse(int x) {
        int reversed = 0;
        while (x != 0) {
            int temp = reversed * 10 + x % 10;
            if ((temp - x % 10) / 10 != reversed) {
                return 0;
            }
            reversed = temp;
            x /= 10;
        }
        return reversed;
    }
}